### ISSUE TITLE
output size error

### ISSUE THREAD

**Issue by MySchizoBuddy** [jeroen/brotli#1]
I'm getting an error while decompressing a 169mb brotli file
`Error in brotli_decompress(buf) : Failed to calculate output size`
This only happens with this particular file which is 1.9GB large list of dataframes in R.  while compressing I used quality=9 

I'm using a function provided by you on your blog

```
save_rds_brotli <- function(object, file, ...){
  buf <- brotli_compress(serialize(object, NULL), ...)
  writeBin(buf, file)
 }

read_rds_brotli <- function(file){
  buf <- readBin(file, raw(), file.info(file)$size)
  unserialize(brotli_decompress(buf))
}
```

I have checked the file size as

```
file.info("data/cleanEPW/epw.sites.data.br")$size
[1] 170003276
```

**Comment by jeroen** [comment]
Might be a bug in upstream. I just updated libbrotli. Can you install the dev version and try again?

``` r
library(devtools)
install_github("jeroenooms/brotli")
```

If this doesn't solve the problem, you need to provide a reproducible example somehow. Maybe try if it also happens for smaller data?

**Comment by MySchizoBuddy** [comment]
The failure occurs at different quality parameter. For smaller dataframe size it fails at lower quality like from 1 to 3. For large dataframes it fails at almost all quality levels. here is the test code

```
 library(brotli)

save_rds_brotli <- function(object, file, ...){
  buf <- brotli_compress(serialize(object, NULL), ...)
  writeBin(buf, file)
}

read_rds_brotli <- function(file){
  buf <- readBin(file, raw(), file.info(file)$size)
  unserialize(brotli_decompress(buf))
}

size.of.df = 200

df <- data.frame(x = replicate(size.of.df, sample(LETTERS, size.of.df * 100, replace = TRUE)))

lapply(c(11:1), function(i) {
  cat("\nCompression with quality ", i)

  save_rds_brotli(df, "df.br", quality = i)
  identical(df, read_rds_brotli("df.br"))

  cat(" Decompression successfull")
})
```

**Comment by jeroen** [comment]
This should be fixed in brotli 0.8 which is on CRAN now. Thanks for reporting this!
